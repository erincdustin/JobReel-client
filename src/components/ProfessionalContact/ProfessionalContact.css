.contact-card {
  width: 80%;
  border: 1px solid black;
  border-radius: 10px;
  margin: 0.75em auto;
  background: #3b3a50;
  color: #eeebd9;
  box-shadow:  0.5em 0.5em 0.2em rgb(48, 39, 39);
  text-align: left;
  padding: 0.5em;
}