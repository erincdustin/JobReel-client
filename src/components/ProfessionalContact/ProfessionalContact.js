import React, { Component } from 'react'
import './ProfessionalContact.css';
export default class ProfessionalContact extends Component {

  renderProfessionalContact() {
    const {professional = {}} = this.props
    return (
    <div className='contact-card'>
      <div className='contact-name'>
        <h4>Name: {professional.first_name} {professional.last_name}</h4>
      </div>
      <div className='contact-department'>
        {professional.department ? <p>Department: {professional.department}</p> : ''}
      </div>
      <div className='contact-position'>
        {professional.position ? <p>Position: {professional.position}</p> : ''}
      </div>
      <div className='contact-type'>
        {professional.type ? <p>Type: {professional.type}</p> : ''}
      </div>
      <div className='contact-email'>
        {professional.value ? <p>Email: {professional.value}</p> : ''}
      </div>
      <div className='contact-phone'>
        {professional.phone_number ? <a href={professional.phone_number} alt='professionals-phone'>Call</a> : ''}
      </div>
      <div>
        {professional.linkedin ?<a href={professional.linkedin} alt='contact-linkedIn-link'>LinkedIn</a> : ''}
      </div>
    </div>
    )
  }

  render() {
    return (
      <>
        {this.renderProfessionalContact()}
      </>
    )
  }
}