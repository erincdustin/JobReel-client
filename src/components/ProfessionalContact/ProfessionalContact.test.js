import React from 'react'
import ReactDOM from 'react-dom'
import ProfessionalContact from './ProfessionalContact'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('ProfessionalContact smoke test', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<ProfessionalContact />, div)
    ReactDOM.unmountComponentAtNode(div)
  })
})

describe('ProfessionalContact snapshot test', () => {
  let props = {
    professional : {
      first_name: 'first',
      department: 'department',
      position: 'position',
      type: 'type',
      value: 'email@email.com',
      phone_number: '111-111-1111',
      linkedin: 'http://linkedin.com/person'
    }
  }

  it('renders the UI as expected', () => {
    const wrapper = shallow(<ProfessionalContact {...props}/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})