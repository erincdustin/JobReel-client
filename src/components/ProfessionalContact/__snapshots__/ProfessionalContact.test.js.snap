// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfessionalContact snapshot test renders the UI as expected 1`] = `
<Fragment>
<<<<<<< HEAD
  <div
    className="contact-card"
  >
    <div
      className="contact-name"
    >
      <h4>
        Name: 
        first
         
      </h4>
    </div>
    <div
      className="contact-department"
    >
      <p>
        Department: 
        department
      </p>
    </div>
    <div
      className="contact-position"
    >
      <p>
        Position: 
        position
      </p>
    </div>
    <div
      className="contact-type"
    >
      <p>
        Type: 
        type
      </p>
    </div>
    <div
      className="contact-email"
    >
      <p>
        Email: 
        email@email.com
      </p>
    </div>
    <div
      className="contact-phone"
    >
      <a
        alt="professionals-phone"
        href="111-111-1111"
      >
        Call
      </a>
    </div>
    <div>
      <a
        alt="contact-linkedIn-link"
        href="http://linkedin.com/person"
      >
        LinkedIn
      </a>
    </div>
=======
  <h2>
    Name: 
    first
     
  </h2>
  <p>
    Department: 
    department
  </p>
  <p>
    Position: 
    position
  </p>
  <p>
    Email: 
    email@email.com
  </p>
  <p>
    LinkedIn: 
    http://linkedin.com/person
  </p>
  <p
    className="error-message"
  />
  <div
    className="save-button"
  >
    <p>
      Saved âœ”
    </p>
>>>>>>> b589b9a987612f9b6a28c9402f9416bf87ef0fcc
  </div>
</Fragment>
`;
